test/types/mongodb.test-d.ts

```diff
@@ -38,6 +39,16 @@ const composedMap = mappedAgg.map<string>(x => x.toString());
 expectType<AggregationCursor<string>>(composedMap);
 expectType<string | null>(await composedMap.next());
 expectType<string[]>(await composedMap.toArray());
+const changeStream = coll.watch();
+changeStream.on('init', (doc) => {
+  expectType<Document>(doc);
+});
+changeStream.on('more', (doc) => {
+  expectType<Document | undefined>(doc);
+});
+changeStream.on('change', (doc) => {
+  expectType<ChangeStreamDocument<Document>>(doc);
```

Comment:
I think we need to test with a schema here to see if the types propagate correctly.

Reply:
Updated.